### PR TITLE
[REVIEW] Update `to_pandas` api docs 

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR updates documentation of `DataFrame.to_pandas` and `Series.to_pandas`.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -4802,6 +4815,25 @@ def to_pandas(self, nullable=False, **kwargs):
...
+        >>> pdf = df.to_pandas(nullable=True)
+        >>> pdf
+              a      b
+        0     0   True
+        1  <NA>  False
+        2     2   <NA>
+        >>> pdf.dtypes
+        a      Int64
+        b    boolean
+        dtype: object
```
Comment (isVoid) [inline python/cudf/cudf/core/dataframe.py:4836]: Addressing the docstring, I think it's also interesting to showcase the situation where `nullable=False`

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -1986,6 +1995,22 @@ def to_pandas(self, index=True, nullable=False, **kwargs):
...
+        1      20
+        2    <NA>
+        3      30
+        dtype: int64
+        >>> ser.to_pandas(nullable=True)
+        0      10
+        1      20
+        2    <NA>
+        3      30
+        dtype: Int64
```
Comment (isVoid) [inline python/cudf/cudf/core/series.py:2013]: Same as above.

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py:4837]: Done

**Review by rgsl888prabhu — APPROVED** [review]

**Review by isVoid — APPROVED** [review]

**Pull request merged**